ggml/src/ggml-hexagon/ggml-hexagon.cpp

```diff
@@ -2560,7 +2384,7 @@ static void ggml_hexagon_binary(const struct ggml_tensor * op, uint32_t flags) {
         (float) sess->prof_cycles / sess->prof_pkts, (unsigned long long) t2 - t1);
 }
 
-static void ggml_hexagon_add_id(const struct ggml_tensor * op, uint32_t flags) {
+template <bool _IsSrc0Constant> static void ggml_hexagon_binary_id(const struct ggml_tensor * op, uint32_t flags) {
```

Comment:
We can probably refactor these op functions into a single ggml_hexagon_generic function that handles null sources. This would reduce code duplication significantly. WDYT? @max-krasnyansky